File: Ser2023/Company/Company.Web/Modules/Company/Currency/CurrencyRow.cs
using Company.Administration.Entities;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace Company.Company;

[ConnectionKey("Company"), Module("Company"), TableName("[sales].[Currency]")]
[DisplayName("Currency"), InstanceName("Currency")]
[ReadPermission(PermissionKeys.Currency.View)]
[ModifyPermission(PermissionKeys.Currency.Modify)]
[DeletePermission(PermissionKeys.Currency.Delete)]
[LookupScript("Company.Currency", LookupType = typeof(MultiTenantRowLookupScript<>))]

public sealed class CurrencyRow : LoggingRow<CurrencyRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Currency Code"), Size(3), PrimaryKey, NotNull, IdProperty, QuickSearch]
    public string CurrencyCode { get => fields.CurrencyCode[this]; set => fields.CurrencyCode[this] = value; }

    [DisplayName("Name"), Size(50), NotNull, NameProperty, QuickSearch]
    public string Name { get => fields.Name[this]; set => fields.Name[this] = value; }

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public Int32Field TenantIdField { get => fields.TenantId; }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    Int16Field IIsActiveRow.IsActiveField { get => fields.IsActive; }

    #endregion Tenant & Activ

    public class RowFields : LoggingRowFields
    {
        public StringField CurrencyCode;
        public StringField Name;

        public Int16Field IsActive;
        public Int32Field TenantId;

    }
}

File: 2026/Company/Company.Web/Modules/ServerTypes/Sales/CurrencyRow.ts
import { fieldsProxy, getLookup, getLookupAsync } from "@serenity-is/corelib";

export interface CurrencyRow {
    CurrencyId?: number;
    CurrencyCode?: string;
    Name?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class CurrencyRow {
    static readonly idProperty = 'CurrencyId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'CurrencyCode';
    static readonly localTextPrefix = 'Company.Currency';
    static readonly lookupKey = 'Sales.Currency';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<CurrencyRow>('Sales.Currency') }
    static async getLookupAsync() { return getLookupAsync<CurrencyRow>('Sales.Currency') }

    static readonly deletePermission = 'Company:Currency:Delete';
    static readonly insertPermission = 'Company:Currency:Modify';
    static readonly readPermission = 'Company:Currency:View';
    static readonly updatePermission = 'Company:Currency:Modify';

    static readonly Fields = fieldsProxy<CurrencyRow>();
}